docs/adr/adr-013-blocksync-overhaul-part-2.md

```diff
@@ -0,0 +1,421 @@
+# ADR #011: Block Data Sync Overhaul: Part II - Network
+
+## Authors
+
+@Wondertan @renenay
+
+> This ADR took a while to get through all the necessary details. Special thanks to awesome @renenay for dozens of
+> deep dive call, note-taking, thoughts and ideas.
+> Similarly, the time was spent on the experiments with the format of the documents was refactored 3 times.
+
+## Glossary
+
+- LN - Light Node
+- FN - Full Node
+- BN - Bridge Node
+- Core Network - Consensus Network of Validators/Proposers producing new Celestia Blocks
+- [EDS(Extended Data Square)](https://github.com/celestiaorg/rsmt2d/blob/master/extendeddatasquare.go#L10) - plain Block
+  data omitting headers and other block metadata
+- ODS - Original Data Square or the first quadrant of the EDS. Contains real user data and padding
+- Recent EDS - Newly produced EDS coming from the Core consensus network
+- ND - Namespaced data/shares. Constitutes EDS
+
+## Overview
+
+This document is a continuation of the BlockSync Overhaul design. The previous document(TODO: Link) described design for the
+storage subsystem of the new block synchronization protocol of Celestia's Data Availability network, while the Part II
+focuses on the networking.
+
+The new design drastically speeds up EDS retrieval in __happy path__ by streaming ODS as a single file and recomputing
+parity data forming EDS. Similar approach is taken for ND retrieval. The main goal is to make EDS and ND retrieval
+consistently outpace EDS production for the max EDS size for historical and recent EDSes.
+
+The existing EDS retrieval logic(TODO: Link something) remains a __fallback path__, if the __happy path__ fails due to lack of connected
+peers or the coordinated data withholding attack. With the __fallback path__ retrieval leverages all connected peers,
+including LNs, and granulary requests EDS pieces - __shares__, increasing the likelihood of EDS reconstruction, while
+paying a slower retrieval time price.
+
+## Decision
+
+The decision is to design three protocols:
+- [`ShrEx/EDS`](#ShrEx/EDS Protocol) - Pull-based EDS retrieval protocol
+- 
+- [`ShrEx/ND`](#ShrEx/ND Protocol) - Pull-based ND retrieval protocol
+- 
+- [`ShrEx/Sub`](#ShrEx/Sub Protocol) - Push-based _notification_ protocol for the newly produced EDSes
+
+### Key Design Decisions
+
+- __Use [go-libp2p][go-libp2p]'s software stack to build the protocols.__ It's extensively used in `celestia-node` already,
+proven to be robust and provides all the requirRed components and tools for the new protocols.
+
+- __Use [protobuf][proto] schema for protocol(TODO Link) message (de)serialization.__ Protobuf is a common (de)serialization
+standard which is already used in celestia-node. Even thought there are better alternatives, their benefits are
```

Comment:
Non blocking comment: better in what sense?

Reply:
Speed, size of data, determinism, not tooling